Bottom: b934e80727956738d70af9d2c8138983137942ac
Top:    7b52dc7e04f3d8fa7e5092bc081a70b3b0a4005a
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-06-04 16:52:30 -0500

Extract buildbot tasks into separate playbook


---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
new file mode 100644
index 00000000..e8629926
--- /dev/null
+++ b/ansible-playbook/buildbot.yml
@@ -0,0 +1,67 @@
+---
+
+- name: Install/Setup Docker images and containers on daedalus
+  hosts: daedalus
+  become: true
+  become_user: root
+  become_method: sudo
+  tags:
+  - buildbot
+  roles:
+  - name: Build and start the Buildbot master container
+    role: buildbot-master
+    tags:
+    - buildbot-master
+
+  - name: Add buildbot user to Gitlab
+    role: gitlab-user
+    vars:
+      new_user: buildbot
+      new_user_name: Build Robot
+      new_user_email: buildbot@novatech-llc.com
+      new_user_pass: "{{ lookup('password', '/dev/null') }}"
+
+- name: Setup Buildbot worker system
+  hosts: buildbot-worker
+  become: true
+  become_user: root
+  become_method: sudo
+  tags:
+  - buildbot
+  - buildbot-worker
+  vars:
+    buildbot_master: "{{ hostvars.buildbot_master.ip_addr }}"
+  roles:
+  - role: buildbot-worker-ntel
+  - role: buildbot-worker-ptxdist
+
+- name: Gitlab public keys for buildbot containers
+  hosts: daedalus
+  become: true
+  become_user: root
+  become_method: sudo
+  tags:
+  - buildbot
+  roles:
+  - name: Add buildbot_master pubkey to buildbot user
+    role: gitlab-sshkey
+    vars:
+      username: buildbot
+      pubkey: "{{ hostvars.buildbot_master.pubkey }}"
+      title: "buildbot_master"
+  - name: Add buildbot_worker_ntel pubkey to buildbot user
+    tags:
+    - buildbot
+    role: gitlab-sshkey
+    vars:
+      username: buildbot
+      pubkey: "{{ hostvars.buildbot_worker_ntel.pubkey }}"
+      title: "buildbot_worker_ntel"
+  - name: Add buildbot_worker_ptxdist pubkey to buildbot user
+    tags:
+    - buildbot
+    role: gitlab-sshkey
+    vars:
+      username: buildbot
+      pubkey: "{{ hostvars.buildbot_worker_ptxdist.pubkey }}"
+      title: "buildbot_worker_ptxdist"
diff --git a/ansible-playbook/site.yml b/ansible-playbook/site.yml
index 82d8ce8c..b43da100 100644
--- a/ansible-playbook/site.yml
+++ b/ansible-playbook/site.yml
@@ -100,11 +100,9 @@
     role: docker-buildsystem
     tags:
     - buildsystem_container
-  - name: Build and start the Buildbot master container
-    role: buildbot-master
-    tags:
-      - buildbot
-      - buildbot-master
+
+- name: Setup the buildbot containers
+  include: buildbot.yml
 
 - name: Setup the Build System (Test Station) with repo data
   hosts: buildsystem
diff --git a/roles/buildbot-master/tasks/main.yml b/roles/buildbot-master/tasks/main.yml
index 94a1884f..ff25c6fb 100644
--- a/roles/buildbot-master/tasks/main.yml
+++ b/roles/buildbot-master/tasks/main.yml
@@ -32,3 +32,15 @@
     volumes: '{{ buildbot_volumes }}'
     env: '{{ buildbot_env }}'
     ports: '{{ buildbot_port_args }}'
+
+- name: Grab ssh public key
+  command: >
+    docker exec '{{ buildbot_container_name }}' cat /home/buildbot/.ssh/id_rsa.pub
+  changed_when: false
+  register: output
+
+- name: Add buildbot_master host
+  add_host:
+    name: buildbot_master
+    pubkey: "{{ output.stdout }}"
+    ip_addr: "{{ buildbot_ip_addr }}"
